chaco/data_view.py

```diff
@@ -208,10 +208,65 @@ class DataView(OverlayPlotContainer):
     bgcolor = "white"
 
     #: Padding defaults.
-    padding_top = 50
-    padding_bottom = 50
-    padding_left = 50
-    padding_right = 50
+    padding_top = Property(
+        observe='y_axis.[title,orientation], x_axis.[title,orientation]'
+    )
+    padding_bottom = Property(
+        observe='y_axis.[title,orientation], x_axis.[title,orientation]'
+    )
+    padding_left = Property(
+        observe='y_axis.[title,orientation], x_axis.[title,orientation]'
+    )
+    padding_right = Property(
+        observe='y_axis.[title,orientation], x_axis.[title,orientation]'
+    )
```

Comment:
I handle all 4 cases because it is possible to set axis orientation to "top" or "bottom" as well

Reply:
as an example, this code:
```
rom numpy import linspace, sin
from traits.api import HasTraits, Instance
from traitsui.api import View, Item
from chaco.api import Plot, ArrayPlotData
from enable.api import ComponentEditor

class LinePlot(HasTraits):
    plot = Instance(Plot)

    traits_view = View(
        Item('plot',editor=ComponentEditor(), show_label=False),
        width=500, height=500, resizable=True, title="Chaco Plot")

    def _plot_default(self):
        x = linspace(-14, 14, 100)
        y = sin(x) * x**3
        plotdata = ArrayPlotData(x=x, y=y)

        plot = Plot(plotdata)
        plot.plot(("x", "y"), type="line", color="blue")
        plot.title = "sin(x) * x^3"

        plot.padding_left = 0
        
        plot.x_axis.title = "X AXIS"
        plot.y_axis.title = "Y AXIS"
        
        plot.y_axis.orientation = "top"
       
        return plot

if __name__ == "__main__":
    LinePlot().configure_traits()
```

yields this on master:
<img width="612" alt="Screen Shot 2021-05-12 at 9 38 47 AM" src="https://user-images.githubusercontent.com/36972686/117994127-eac63a00-b305-11eb-9772-cf7dcd6248a4.png">

and this on this branch:
<img width="612" alt="Screen Shot 2021-05-12 at 9 39 02 AM" src="https://user-images.githubusercontent.com/36972686/117994158-f0bc1b00-b305-11eb-8ab0-639c80f8cf36.png">

Reply:
The extra padding on the bottom on this branch (80 rather than 50) I'm not a huge fan of.  This is due to the simplistic nature of just adding space if we have an axis title.  As you can see, at the top we need extra space so axis title doesn't overlap with plot title.  Similarly left right we need it (hence the original issue), because axis ticks / values take up some of the spacing and sort of push the axis title further away from the plot.  The x axis doesn't usually have this problem because of the orientation of the numbers on the ticks.  In other words the text for `1000` for example is much wider than it is tall, and `100000` even wider, but the same height.

In reality an ideal solution would calculate explicitly the extent of the ticks, and the size of the axis title and allocate spacing to perfectly fit what we need.  The awkward thing about this code feels like it would live in a draw method inside the axis itself, but the DataView is where the padding is held.  Should the Axis take control over the DataViews padding?  That feels wrong.
In any case that is a separate issue / approach entirely.  I am just pointing out something I don't love about the current solution.

Perhaps we just make the assumption that padding_bottom can be 50 regardless of the presence or absence of an axis title?  We were just assuming 50 always before anyway.